code/modules/mob/living/carbon/life.dm

```diff
@@ -314,7 +314,10 @@
 /mob/living/carbon/handle_status_effects()
 	..()
 	if(staminaloss)
-		adjustStaminaLoss(-3)
+		if(IsKnockdown() || IsUnconscious())

```

Comment:
In general, you shouldn't be checking for these in Life() or its subprocs at all. That's why they're status effects.